src/libs/text-selection/text-selection-control.js

```diff
@@ -0,0 +1,75 @@
+define("libs/text-selection/text-selection-control", function(require, exports, module) {
+
+	// imports
+	var Event = require("libs/events/event");
+	var EventDispatcher = require("libs/events/event-dispatcher");
+	var TextSelection = require("libs/text-selection/text-selection");
+
+	/**
+	 * Control for text selection on page.
+	 */
+	var TextSelectionControl = Class(EventDispatcher, {
+
+		/**
+		 * Helper constructor.
+		 */
+		constructor : function () {
+			TextSelectionControl.$super.call(this);
+
+			this._selection = new TextSelection();
+
+			this._onMouseUpHandler = this._onMouseUp.bind(this);
+			this._onMouseOutHandler = this._onMouseOut.bind(this);
+			this._onKeyUpHandler = this._onKeyUp.bind(this);
+
+			document.addEventListener("mouseup", this._onMouseUpHandler);
```

Comment:
I've tried, but problem is that constants are in shared directory and it represents constant used in app not in libs. All app code depends on libs so shared and other app code should not be used in (included) in libs.